### ISSUE TITLE
Limited number of functioning weapons on modded ships

### ISSUE THREAD

**Issue by FIM43-Redeye** [endless-sky/endless-sky#5511]
Ships seem to only support so many installed weapons, probably around 15 or so, not counting anti-missile turrets. When I removed the gun ports from a ship, the turrets started working, but again, they were limited significantly.

1. Create a modded ship with a large number of guns or turrets (overlapping works)
2. Attempt to fire the weapons

Expected behavior:
All turrets would fire in concert, as well as forward-mounted guns.

Screenshots:
https://puu.sh/GPB7Q/cdfc859c8e.png

System:
 - OS: Windows 10 Enterprise, version 2004, build 19041.630]
 - Game Source: Steam
 - Version: 0.9.12

Link to save file:
[Abigail Steele.txt](https://github.com/endless-sky/endless-sky/files/5576664/Abigail.Steele.txt)

Additional context:
Yes, I know I'm a filthy cheater, but also it's a legitimate bug. Modern PCs are very strong so I don't expect this is a performance issue.

**Comment by Anarchist2** [comment]
#4635

**Comment by Zitchas** [comment]
As Anarchist2's post suggests, there is currently a hard limit of 32 weapons in the game. Any more beyond that simply won't function. #4635 will fix that, but it hasn't been merged yet.

**Comment by FIM43-Redeye** [comment]
Alright ,that makes sense. I am incredibly new to Github in general, do you have any idea how #4635 will get merged, how it should be merged, etc.?

**Comment by MCOfficer** [comment]
TL;DR: eventually, relax and wait.

First, reviewers have to go though it and add suggestions, and the PR author has to apply them. When that ping pong is done and a maintainer with sufficient coding skills (read: tehhowch) is satisfied, it can be merged - though it will likely be merged into an unstable release (uneven version number) rather than a stable one. Unless you have or develop sufficient coding skills to take part in the ping pong, all you can do is subscribe and wait.

**Comment by FIM43-Redeye** [comment]
That makes sense. Thanks for the explanation, I really appreciate it.
Closed.
